homeassistant/components/toon/switch.py

```diff
@@ -0,0 +1,121 @@
+"""Support for Toon switches."""
+import logging
+from typing import Any
+
+from toonapi import (
+    ACTIVE_STATE_AWAY,
+    ACTIVE_STATE_HOLIDAY,
+    PROGRAM_STATE_OFF,
+    PROGRAM_STATE_ON,
+)
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    ATTR_DEFAULT_ENABLED,
+    ATTR_ICON,
+    ATTR_INVERTED,
+    ATTR_MEASUREMENT,
+    ATTR_NAME,
+    ATTR_SECTION,
+    DOMAIN,
+    SWITCH_ENTITIES,
+)
+from .coordinator import ToonDataUpdateCoordinator
+from .helpers import toon_exception_handler
+from .models import ToonDisplayDeviceEntity, ToonEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up a Toon switches based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [ToonProgramSwitch(coordinator), ToonHolidayModeSwitch(coordinator)]
```

Comment:
Why are these not climate presets ?

Reply:
The program is a schedule, a preset may override the schedule.
Holiday mode is some kind of preset for Toon (it is hidden in the API as preset as well), and from Home Assistant perspective, it makes sense. However, it does act differently (turns of the display, disables a bunch of features, presets become unresponsive).

Turning it on/off needs different API calls as well. Hence, I've split it into a switch. I think the nice part of it, is that it becomes a nice switch for HA itself as well to know when it is holiday mode (as a general broader use case).